flink-staging/flink-streaming/flink-storm-compatibility/src/main/java/org/apache/flink/stormcompatibility/wrappers/StormWrapperSetupHelper.java

```diff
@@ -0,0 +1,123 @@
+	 *            Set to {@code true} if a single attribute output stream, should not be of type {@link Tuple1} but be
+	 *            of a raw type.
+	 * @return The number of attributes to be used.
+	 * @throws IllegalArgumentException
+	 *             If {@code rawOuput} is {@code true} and the number of declared output attributes is not 1 or if
+	 *             {@code rawOuput} is {@code false} and the number of declared output attributes is not with range
+	 *             [1;25].
+	 */
+	public static int getNumberOfAttributes(final IComponent spoutOrBolt, final boolean rawOutput)
+		throws IllegalArgumentException {
+		final StormOutputFieldsDeclarer declarer = new StormOutputFieldsDeclarer();
+		spoutOrBolt.declareOutputFields(declarer);
+		
+		final int declaredNumberOfAttributes = declarer.getNumberOfAttributes();
+		
+		if(declaredNumberOfAttributes == -1) {
+			return -1;
+		}
+		
+		if((declaredNumberOfAttributes < 1) || (declaredNumberOfAttributes > 25)) {
+			throw new IllegalArgumentException(
+				"Provided bolt declares non supported number of output attributes. Must be in range [1;25] but was "
+					+ declaredNumberOfAttributes);
+		}
+		
+		if(rawOutput) {
+			if(declaredNumberOfAttributes > 1) {
+				throw new IllegalArgumentException(
+					"Ouput type is requested to be raw type, but provided bolt declares more then one output attribute.");
+				
+			}
+			return 0;
+		}
+		
+		return declaredNumberOfAttributes;
+	}
+	
+	/**
+	 * TODO
+	 */
+	public static TopologyContext convertToTopologyContext(final StreamingRuntimeContext context, final boolean spoutOrBolt) {
+		final Integer taskId = new Integer(1 + context.getIndexOfThisSubtask());
+		
+		final Map<Integer, String> taskToComponents = new HashMap<Integer, String>();
+		taskToComponents.put(taskId, context.getTaskName());
+		
+		final ComponentCommon common = new ComponentCommon();
+		common.set_parallelism_hint(context.getNumberOfParallelSubtasks());
+		
+		final Map<String, Bolt> bolts = new HashMap<String, Bolt>();
+		final Map<String, SpoutSpec> spoutSpecs = new HashMap<String, SpoutSpec>();
+		
+		if(spoutOrBolt) {
+			spoutSpecs.put(context.getTaskName(), new SpoutSpec(null, common));
+		} else {
+			bolts.put(context.getTaskName(), new Bolt(null, common));
+		}
+		
+		// FlinkTopologyContext(StormTopology topology, Map stormConf, Map<Integer, String> taskToComponent,
+		// Map<String, List<Integer>> componentToSortedTasks, Map<String, Map<String, Fields>>
+		// componentToStreamToFields, String stormId, String codeDir, String pidDir, Integer taskId, Integer
+		// workerPort, List<Integer> workerTasks, Map<String, Object> defaultResources, Map<String, Object>
+		// userResources, Map<String, Object> executorData, Map registeredMetrics, Atom openOrPrepareWasCalled)
+		
+		return new FlinkTopologyContext(new StormTopology(spoutSpecs, bolts, null), null, taskToComponents, null, null,
+			null, null, null, taskId, null, null, null, null, null, null, null);
+	}
```

Comment:
TopologyContext is not supported so far. This is only a pre-alpha version to make it compile and run. Added TopologyContext (in a nice way) need some more work. Not sure If we need it for the first pull request?